app/code/Magento/Config/Plugin/App/Config/Type/System/ChangeConfigPath.php

```diff
@@ -0,0 +1,158 @@
+ */
+class ChangeConfigPath
+{
+    /**
+     * @var ScopeCodeResolver
+     */
+    private $scopeCodeResolver;
+
+    /**
+     * Deployment configuration
+     *
+     * @var DeploymentConfig
+     */
+    private $deploymentConfig;
+
+    /**
+     * The resolver for configuration paths according to source type.
+     *
+     * @var ConfigPathResolver
+     */
+    private $configPathResolver;
+
+    /**
+     * @var array
+     */
+    private $websiteCode = [];
+
+    /**
+     * Config constructor.
+     *
+     * @param ScopeCodeResolver $scopeCodeResolver
+     * @param DeploymentConfig $deploymentConfig
+     * @param ConfigPathResolver $configPathResolver
+     */
+    public function __construct(
+        ScopeCodeResolver $scopeCodeResolver,
+        DeploymentConfig $deploymentConfig,
+        ConfigPathResolver $configPathResolver
+    ) {
+        $this->scopeCodeResolver = $scopeCodeResolver;
+        $this->deploymentConfig = $deploymentConfig;
+        $this->configPathResolver = $configPathResolver;
+    }
+
+    /**
+     * Before plugin for change config path base deployment config
+     *
+     * @param System $subject
+     * @param string $path
+     * @return array
+     */
+    public function beforeGet(
+        System $subject,
+        $path = ''
+    ) {
+        $pathParts = explode('/', $path);
+        $scopeType = array_shift($pathParts);
+        $scopeId = array_shift($pathParts);
+        if ($path === '' || $scopeType === ScopeInterface::SCOPE_DEFAULT) {
+            return [$path];
+        }
+
+        $configPath = implode('/', $pathParts);
+
+        if ($scopeType === StoreScopeInterface::SCOPE_STORES) {
+            $websiteCode = $this->getWebsiteCodeFromStore($scopeType, $this->getScopeCode($scopeType, $scopeId));
+            if ($websiteCode) {
+                if ($this->isLocked($configPath, StoreScopeInterface::SCOPE_WEBSITES, $websiteCode)) {
+                    array_unshift($pathParts, StoreScopeInterface::SCOPE_WEBSITES, $websiteCode);
+                    $path = implode('/', $pathParts);
+                } elseif ($this->isLocked($configPath, ScopeConfigInterface::SCOPE_TYPE_DEFAULT, null)) {
+                    array_unshift($pathParts, ScopeConfigInterface::SCOPE_TYPE_DEFAULT);
+                    $path = implode('/', $pathParts);
+                }
+            }
+        } elseif ($this->isLocked($configPath, ScopeConfigInterface::SCOPE_TYPE_DEFAULT, null)) {
+            array_unshift($pathParts, ScopeConfigInterface::SCOPE_TYPE_DEFAULT);
+            $path = implode('/', $pathParts);
+        }
+
+        return [$path];
+    }
+
+    /**
+     * Checks whether configuration is locked in file storage.
+     *
+     * @param string $path The path to configuration
+     * @param string $scope The scope of configuration
+     * @param string $scopeCode The scope code of configuration
+     * @return bool
+     */
+    private function isLocked($path, $scope, $scopeCode)
```

Comment:
Use strict type here
```suggestion
    private function isLocked(string $path, string $scope, string $scopeCode)
```

Reply:
Add return type as well
```suggestion
    private function isLocked($path, $scope, $scopeCode): bool
```